src/main/java/it/gov/pagopa/pu/registry/mapper/DebtPositionSendEventDTOMapper.java

```diff
@@ -0,0 +1,35 @@
+package it.gov.pagopa.pu.registry.mapper;
+
+import it.gov.pagopa.pu.registry.dto.KeyValueDTO;
+import it.gov.pagopa.pu.registry.event.payments.dto.DebtPositionSendEventDTO;
+import it.gov.pagopa.pu.registry.utils.StringBuilderUtils;
+import org.springframework.util.StringUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DebtPositionSendEventDTOMapper {
+
+  public String getEventDescription(DebtPositionSendEventDTO dto) {
+    StringBuilder sb = new StringBuilder();
+
+    if (StringUtils.hasText(dto.getEventDescription())) {
+      sb.append(dto.getEventDescription());
+    }
+
+    List<KeyValueDTO<String>> additionalInformationArray = new ArrayList<>();
+
+    if (dto.getPayload() != null && dto.getPayload().getIun() != null) {
+      additionalInformationArray.add(new KeyValueDTO<>("IUN", dto.getPayload().getIun()));
+    }
+
+    if (dto.getPayload() != null && dto.getPayload().getNotificationDate() != null) {
+      additionalInformationArray.add(new KeyValueDTO<>("NotificationDate", dto.getPayload().getNotificationDate().toString()));
+    }
+
+    StringBuilderUtils.appendFromArray(sb, additionalInformationArray, ": ", "; ");
+
+    return sb.toString();
```

Comment:
dto.getPayload cannot be null, moreover this could be simplified:
```suggestion
    return Stream.of(
          dto.getEventDescription(),
          Optional.ofNullable(dto.getPayload().getIun()).map(iun -> "IUN:" + iun).orElse(null),
          Optional.ofNullable(dto.getPayload().getNotificationDate()).map(dt -> "NotificationDate:" + dt).orElse(null),
       )
       .filter(Objects::nonNull)
       .collect(Collectors.joining(";"));
```